torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -1141,31 +1224,48 @@ def call_triton_kernel(self, variable, args, kwargs, tx):
 
 
 class TracingTritonHOPifier(TritonHOPifier):
-    def raise_unsupported(self, msg):
+    def raise_unsupported(self, msg: str) -> None:
         raise RuntimeError(msg)
 
-    def is_callable(self, maybe_callable):
+    def is_callable(self, maybe_callable: Any) -> bool:
         return callable(maybe_callable)
 
-    def get_value(self, val):
+    def get_value(self, val: Any) -> Any:
         return val
 
-    def call_grid(self, grid, meta, tx):
+    def call_grid(
+        self,
+        grid: "Union[TritonGridCallableType, VariableTracker]",
+        meta: "Union[TritonMetaParamsType, VariableTracker]",
```

Comment:
slightly weird because these can't be VariableTracker

Reply:
@aorenste can I ask for your input on what to do here?

TritonHOPifier is the base class; and then DynamoTritonHOPifier and TracingTritonHOPifier are subclasses. Based on their implementations, I imagine their call_grid signatures could be (i.e. mirroring the snippet in another one of your comments):

```
TracingTritonHOPifier:  # call this "signature A"
    def call_grid(self, grid: TritonGridCallableType, meta: TritonMetaParamsType, tx: None) -> Tuple[Union[int, sympy.Expr, SymInt], ...]:

DynamoTritonHOPifier:    # call this "signature B"
    def call_grid(self, grid: VariableTracker, meta: VariableTracker, tx: InstructionTranslator) -> Tuple[Proxy, ...]:

TritonHOPifier:  # call this "signature C"
    def call_grid(self, grid: VariableTracker | TritonGridCallableType, meta: VariableTracker | TritonMetaParamsType, tx: Optional[InstructionTranslator]) -> Union[Tuple[Union[int, sympy.Expr, SymInt], ...], Tuple[Proxy, ...]]:
```

I can mark TritonHOPifier with "signature C", or with the `@overload` strategy you mentioned.

But then when I add types to (Tracing|Dynamo)TritonHOPifier, mypy requires the signatures of `call_grid` implementations on these classes to match (at least for the parameters; and in the case of using `@overload`, I need to copy the `@overload` declarations to the subclasses).

Any ideas on a cleaner way to add typing in this scenario?

Reply:
okay, maybe I'll just not annotate the types on the base class and mark those as `type: ignore[no-untyped-def]`. That seems to work.

Reply:
I think the problem here is that this API is just bad. I don't know the details other than what you said here but it sounds like TracingTritonHOPifier and DynamoTritonHOPifier's call_grid() are just unrelated functions that happen to have the same name? Do we actually need to define it on TritonHOPifier? Like, does anyone actually have a plain old TritonHOPifier and expect to call `call_grid` on it? Without knowing which specific subtype they have how do they know what to pass it?

I suspect that for now your plan of not annotating the base class (or use 'Any' just to indicate that someone has at least kind of thought about it and bailed) is probably reasonable for now.

Reply:
Thanks for the input! I'll keep it as is then (with no annotations on the base class)

>  Like, does anyone actually have a plain old TritonHOPifier and expect to call call_grid on it?

No... 

> it sounds like TracingTritonHOPifier and DynamoTritonHOPifier's call_grid() are just unrelated functions that happen to have the same name

Well, logically they do the same thing, but one operates on actual grid objects, while the other one operates on dynamo VariableTrackers that represent grid objects...

Reply:
> Thanks for the input! I'll keep it as is then (with no annotations on the base class)
> 
> > Like, does anyone actually have a plain old TritonHOPifier and expect to call call_grid on it?
> 
> No...

In which case (in the future) maybe the right thing is to just remove `call_grid` from the base class.